### PR TITLE
Fix service expected fieldOfActivityId behaviour.

### PR DESCRIPTION
**Pull request by laurandarius**

### CODE REVIEW

**Comment by laurandarius** [comment]
[Company service expects "fieldOfActivityId" property but "business" is provided.](https://trello.com/c/u1N7ZfWA/17-company-service-expects-fieldofactivityid-property-but-business-is-provided)

File: app/services/company.js [hunk 1]
```diff
@@ -15,28 +16,38 @@ class CompanyService extends CrudService {
       error.throwValidationError('Company already exists.');
     }
 
-    if (company.fieldOfActivityId !== undefined) {
-      const fieldExists = await fieldOfActivityService.get(company.fieldOfActivityId);
+    const { business, ...newCompany } = company;
+    if (business !== undefined) {
+      const fieldExists = await fieldOfActivityRepository.getByPropsNonParanoid({ name: business });
```
Comment (lgnstn) [inline app/services/company.js]: This is not the best name for this variable. Here I consider it a boolean value, but a few lines below I see that it is in fac an object. I would call it "fieldOfActivity".

File: app/services/company.js [hunk 2]
```diff
@@ -15,28 +16,38 @@ class CompanyService extends CrudService {
...
-    if (newData.name !== undefined) {
+    if (data.name !== undefined) {
       error.throwValidationError('You can not change the company name.');
     }
 
-    if (newData.fieldOfActivityId !== undefined) {
-      const fieldExists = await fieldOfActivityService.get(newData.fieldOfActivityId);
+    const { business, ...newData } = data;
+    if (business !== undefined) {
+      const fieldExists = await fieldOfActivityRepository.getByPropsNonParanoid({ name: business });
```
Comment (lgnstn) [inline app/services/company.js]: Same here.

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by laurandarius** [comment]
@lgnstn updated.

**Review by lgnstn — APPROVED** [review]

**Comment by lgnstn** [comment]
@laurandarius Conflict.

**Comment by laurandarius** [comment]
@lgnstn updated.

**Pull request merged**